packages/react-native-payments/lib/js/PaymentRequest.js

```diff
@@ -287,41 +299,33 @@ export default class PaymentRequest {
     this._shippingOptionChangeFn(event);
   }
 
-  _getPlatformDetails(details) {
+  _getPlatformDetails(details: *) {
     return IS_IOS
       ? this._getPlatformDetailsIOS(details)
       : this._getPlatformDetailsAndroid(details);
   }
 
-  _getPlatformDetailsIOS(details: {
-    transactionIdentifier: string,
-    paymentData: string
-  }) {
+  _getPlatformDetailsIOS(details: PaymentDetailsIOSRaw): PaymentDetailsIOS {
     const {
+      paymentData: serializedPaymentData,
+      paymentToken,
       transactionIdentifier,
-      paymentData: serializedPaymentData
     } = details;
-    const isSimulator = transactionIdentifier === 'Simulated Identifier';
 
-    if (isSimulator) {
-      return Object.assign({}, details, {
-        paymentData: null,
-        serializedPaymentData
-      });
-    }
+    const isSimulator = transactionIdentifier === 'Simulated Identifier';
 
     return {
+      paymentData: isSimulator ? null : JSON.parse(serializedPaymentData),
```

Comment:
I could see this stumping some people.  Could you update the README to warn users of the behaviour in TEST.

Reply:
Sure, documentation is improved.